api/go.mod

```diff
@@ -3,23 +3,24 @@ module github.com/openstack-k8s-operators/keystone-operator/api
 go 1.19
 
 require (
+	github.com/go-logr/logr v1.2.4
+	github.com/google/uuid v1.3.1
+	github.com/onsi/gomega v1.27.10
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.1-0.20230919113507-d74c2f31d216
 	github.com/openstack-k8s-operators/lib-common/modules/openstack v0.1.1-0.20230919113507-d74c2f31d216
+	github.com/openstack-k8s-operators/lib-common/modules/test v0.1.1-0.20230919113507-d74c2f31d216
```

Comment:
We need to consider if this is an acceptable dependency growth in the /api module or not. If it is considered too much then we can put the code into /test/helpers where "helpers" will be a new go.mod to import.

If every project that depends on keystone-operator/api today also expected to have some EnvTest coverage then this dependency increase is not a big deal as those project needs these deps anyhow to implement the tests. So the real question is: are there any projects that depends on (or will depend on) keystone/api but does not want / need EnvTest coverage? Because for them this might be an unnecessary set of dependencies.

Reply:
As the one who has probably been the most outspoken about the dependency growth here I think this is acceptable. One question on modules/test though below...

Reply:
For now I agree its ok. Depending on what is required in the test helpers in the future it could grow and we'd have to make a decision later when its to much to not be in /api module.